packages/typespec-ts/test/modularUnit/operations.spec.ts

```diff
@@ -4,6 +4,92 @@ import { assertEqualContent } from "../util/testUtil.js";
+      assert.ok(operationFiles);
+      assert.equal(operationFiles?.length, 1);
+      await assertEqualContent(
+        operationFiles?.[0]?.getFullText()!,
+        `
+        import { TestingContext as Client } from "../rest/index.js";
+        import { StreamableMethod, operationOptionsToRequestParameters, createRestError } from "@azure-rest/core-client";
+
+        export function _readSend(context: Client, options: ReadOptionalParams = { requestOptions: {} }): StreamableMethod<Read204Response> {
+          return context.path("/", ).post({...operationOptionsToRequestParameters(options)})  ;
+        }
+
+        export async function _readDeserialize(result: Read204Response): Promise<void> {
+          if(result.status !== "204"){
+            throw createRestError(result);
+          }
+      
+          return;
+        }
+        
+        export async function read(context: Client, options: ReadOptionalParams = { requestOptions: {} }): Promise<void> {
+            const result = await _readSend(context, options);
+            return _readDeserialize(result);
+        }`
+      );
+    });
+
+    it("void response body should be omitted", async () => {
+      const tspContent = `
+        op read(): { @body _: void;};
+          `;
+
+      const operationFiles =
+        await emitModularOperationsFromTypeSpec(tspContent);
+      assert.ok(operationFiles);
+      assert.equal(operationFiles?.length, 1);
+      await assertEqualContent(
+        operationFiles?.[0]?.getFullText()!,
+        `
+        import { TestingContext as Client } from "../rest/index.js";
+        import { StreamableMethod, operationOptionsToRequestParameters, createRestError } from "@azure-rest/core-client";
+
+        export function _readSend(context: Client, options: ReadOptionalParams = { requestOptions: {} }): StreamableMethod<Read204Response> {
+          return context.path("/", ).get({...operationOptionsToRequestParameters(options)})  ;
+        }
+
+        export async function _readDeserialize(result: Read204Response): Promise<void> {
+          if(result.status !== "204"){
+            throw createRestError(result);
+          }
+      
+          return;
+        }
+        
+        export async function read(context: Client, options: ReadOptionalParams = { requestOptions: {} }): Promise<void> {
+            const result = await _readSend(context, options);
+            return _readDeserialize(result);
+        }`
+      );
+    });
+
+    it("should throw exception if property type as void", async () => {
+      try {
+        const tspContent = `
+        model Foo {
+          param: void;
+        }
+        op read(...Foo): {};
+          `;
+
+        await emitModularOperationsFromTypeSpec(tspContent);
+        assert.fail("Should throw diagnostic errors");
+      } catch (e: any) {
+        assert.equal(e[0]?.code, "@azure-tools/typespec-ts/invalid-schema");
+        assert.equal(e[0]?.message, "Couldn't get schema for type Intrinsic");
```

Comment:
can this error message be more specific?

Reply:
there are type relevant actually and i could add the assertion here.

Reply:
l leverage the existing rule `invalid-schema` for this check and let me know if you think we need a new linter for void.